src/apps/watchfaces/OswAppWatchfaceDigital.cpp

```diff
@@ -60,22 +61,24 @@ void OswAppWatchfaceDigital::dateOutput(uint32_t yearInt, uint32_t monthInt, uin
 }
 
 static void drawDate(time_t timeZone, uint8_t fontSize, uint8_t CoordY) {
-    uint32_t dayInt = 0;
-    uint32_t monthInt = 0;
-    uint32_t yearInt = 0;
+    OswDate oswDate = { };
     OswHal* hal = OswHal::getInstance();
-    const char* weekday = hal->getWeekday(timeZone);
 
-    hal->getDate(timeZone,&dayInt, &monthInt, &yearInt);
+    hal->getDate(timeZone, oswDate);
 
     // we want to output a value like "Wed, 05/02/2021"
 
     hal->gfx()->setTextSize(fontSize);
     hal->gfx()->setTextMiddleAligned();
     hal->gfx()->setTextLeftAligned();
-    hal->gfx()->setTextCursor(120 - hal->gfx()->getTextOfsetColumns(6.9), CoordY);
+    hal->gfx()->setTextCursor(120 - hal->gfx()->getTextOfsetColumns(6.9f), CoordY);
 
-    OswAppWatchfaceDigital::displayWeekDay3(weekday);
+    try {
+        const char* weekday = hal->getWeekDay.at(oswDate.weekDay);
+        OswAppWatchfaceDigital::displayWeekDay3(weekday);
+    } catch (const std::out_of_range& ignore) {
+        OSW_LOG_E("getWeekDay is out of range: ", ignore.what());
+    }
```

Comment:
It seems highly unlikely, but in the rare case of an exceptional error, the "day of the week" will not be displayed, and it will proceed directly to display the "date."

Reply:
...which is better than crashing? Or what is the question?